quint/src/types/base.ts

```diff
@@ -29,6 +29,7 @@ export interface TypeScheme {
 
 export type Signature = (_arity: number) => TypeScheme
 
+// TODO: Why aren't we noting type variables in the schema here?
```

Comment:
Not sure I understand the question, but this `toScheme` is just a helper to convert a `QuintType` into a `TypeScheme` with no quantified variables.

Reply:
But why would we drop the type variables from a quint type if it had them?

Reply:
`typeVariables` in `TypeScheme` corresponds only to universally quantified type variables, not all type variables. Here's one example where we decide what to quantify:

https://github.com/informalsystems/quint/blob/main/quint/src/types/constraintGenerator.ts#L419

and the `toScheme` is used when we don't want to quantify anything.

Reply:
I see. Thank you for clarifying! I'll replace this comment with something describing that logic. :pray:

Reply:
Actually, I have a followup :)

Should this only be applied to types which have no free variables? Or is it valid to apply this to an open term?

Reply:
I think so! We should probably enforce that requirement. I'm not 100% sure tho. The requirement makes sense, and looking at its usages on the code, I'm like 80% sure they respect this requirement.

Reply:
Cool. I added a comment to this effect. Thanks for explaining!